[den] Week1 학습계획

sjuhan123 opened the pull request:
## 학습계획

**:one: 월요일 학습계획**

- [ ] : 개념 학습
  - [ ] : 디자인 패턴 학습 - 컴퍼넌트 단위 개발? mvc? 
  - [x] : css grid 학습
  - [x] : async/await 개념 학습

**:two: 화요일 학습계획**

- [ ] : 프로그래밍 설계
  - [ ] : feature list 작성
  - [ ] : 설계서 작성
- [ ] : 초기 환경 세팅
  - [x] : prettier, eslint
  - [x] : package.json

## 설계도

<img width="710" alt="스크린샷 2023-04-05 오후 7 54 21" src="https://user-images.githubusercontent.com/81420856/230060399-16631486-e8f1-4bc0-bab6-2bc9472f2a91.png">

## 예상 로직 흐름도

<img width="1011" alt="스크린샷 2023-04-05 오후 8 04 58" src="https://user-images.githubusercontent.com/81420856/230062658-da7b09c8-bd00-4264-aa25-01f1cff6e46b.png">

## 구현 방향
- 중복 사용되는 로직(메서드 등)을 최상위 부모 클래스에서 가지고 있게 해서 중복되는 로직을 줄이는 방향으로 구현할 예정입니다.
- 비동기 처리하는 로직을 분리해서, 비동기 처리와 받아온 데이터를 가공하는 로직을 app 로직에서 분리할 예정입니다.

## 처음 적용한 환경 세팅
- gitignore
- prettier, eslint

## 처음 접하거나 프로젝트에 처음 사용할 예정인 개념  
- 클래스 필드 - 인스턴스 메서드&변수, public / private / protected
- 전개구문 - 객체 간 연결하거나 병합시
- class 상속, 메소드 오버라이딩
- 웹 컴포넌트 바닐라 js로 구현하기 - 상태 관리, diff 알고리즘

## 고민중인 부분
- 최초 렌더링 이후에 이벤트가 발생할 때 마다 해당 컴퍼넌트가 리렌더링이 됩니다. 변경되는 부분만 리렌더링이 되지 않고, 전체가 리렌더링이 될 것 같아, 상태 관리와 최적화를 어떻게 해야 될지 고민이 됩니다. store 개념과 diff 알고리즘이라는 개념이 있던데, 관련 개념을 더 공부해보고 코드에 적용해보겠습니다.

## 앞으로 고민해야 되는 부분
- 웹 컴포넌트를 사용해야 되는 이유와 사용했을 때 사용자 혹은 개발자에게 주는 장,단점 고민해보기
- 컴포넌트를 분리하는 이유 중 하나가 재사용성인데, 내 코드가 정말 재사용이 가능한지 고민해보기

sjuhan123 committed:
docs: 월요일, 화요일 학습계획 작성

sjuhan123 committed:
feat: setup build configuration

sjuhan123 committed:
image: icon 이미지 추가

sjuhan123 committed:
feat: 환경 설정 추가

sjuhan123 committed:
docs: planner 추가

sjuhan123 committed:
feat: 바닐라라js로 웹컴퍼넌트 만들기 연습 겸 초기 상태만 세팅 진행

1. app 컨테이너 생성
2. app 컨테이너 각 섹션 div 생성
3. Header 컴퍼넌트 생성
4. newsstand-trend 컴퍼넌트 생성

caesar1030 commented on package.json:
```diff
@@ -0,0 +1,12 @@
+    "prettier": "^2.8.7"
+  },
+  "scripts": {
+    "watch": "json-server --watch db.json "
```
scripts에서 watch 옵션은 워낙 많이 줘서 
좀더 명확한 이름으로 바꾸면 좋을 것 같아요 server 라던지

sjuhan123 committed:
docs: planner w01/final 계획 작성

sjuhan123 committed:
feat: json server install

sjuhan123 committed:
refactor: db data 변경

sjuhan123 committed:
feat: view container grid view 1차 구현 완료

1. view container용 data 불러오는 비동기 처리 구현
2. view container 부모 컨테이너 html 구현
3. view container grid view 컨포넌트 구현
4. grid view 컨포넌트 페이징 이벤트 구현

sjuhan123 committed:
feat: 기초 css 1차 구현

sjuhan123 committed:
feat: crawl 로직 구현 진행 중

sjuhan123 committed:
feat: grid view grid item mouseover/leave 이벤트 등록 기능 추가

sjuhan123 committed:
delete: 필요없는 주석 삭제

crongro commented:
> * 최초 렌더링 이후에 이벤트가 발생할 때 마다 해당 컴퍼넌트가 리렌더링이 됩니다. 변경되는 부분만 리렌더링이 되지 않고, 전체가 리렌더링이 될 것 같아, 상태 관리와 최적화를 어떻게 해야 될지 고민이 됩니다. store 개념과 diff 알고리즘이라는 개념이 있던데, 관련 개념을 더 공부해보고 코드에 적용해보겠습니다.

개발하다보면 늘 생기는 고민인데요. 해결방법은 분기처리해서 그때그때 렌더링을 결정하는것인데 쉽지 않죠.
꼭 고민해볼만한 주제라고 생각해요.

crongro commented on package.json:
```diff
@@ -0,0 +1,14 @@
+{
+  "dependencies": {
+    "eslint": "^8.37.0",
+    "eslint-config-prettier": "^8.8.0",
+    "eslint-plugin-prettier": "^4.2.1",
+    "prettier": "^2.8.7"
+  },
```
devDependencies 로 옮겨야할듯.
두 개 차이는 뭘까요?

crongro commented on src/app.js:
```diff
@@ -0,0 +1,31 @@
+  async mounted() {
+    const header = this.target.querySelector(".newsstand-header");
+    const trendNews = this.target.querySelector(".newsstand-trendnews");
+    const view = this.target.querySelector(".newsstand-main");
```
변수이름은 view보다는 main이나 mainView?

crongro commented on src/app.js:
```diff
@@ -0,0 +1,31 @@
+
+    const pressData = await getPressData();
+
+    new Header(header);
+    new TrendNews(trendNews);
```
이 두개는 pressData 와 상관이 초기화 되는것이면 await 전에 표현해도 될 듯.

crongro commented on src/core/Component.js:
```diff
@@ -0,0 +1,23 @@
+  mounted() {}
+  setEvent() {}
+  render() {
+    this.target.innerHTML = this.templete();
```
render 가 파라미터 받는것도 없고 무조건  innerHTML 로 넣어버리니 이 부분의 개선이 필요하긴 하겠네요.
쉬운 방법은 없을거에요..지저분하거나.. 어렵거나..등등

crongro reviewed (approved):
뷰 클래스 형태가 일관되어 있고 깔끔하군요.
지금 패턴 잘 연구해서 본인의 것으로 만드시고, 장단점을 따져가며 비판적으로 사용하시길 바랄게요.
